### PR TITLE
Fixes part of #2322: Changes in Previous Responses Header Item [A11y]

### PR DESCRIPTION
**Pull request by Mehul-Bisht**
## Explanation

  
Fixes part of #2322: Changes in Previous Responses Header Item [A11y]  
This PR fixes the accessibility issue for the height of Previous Responses Header in mobile layouts only, since I was not able to set up the accessibility scanner on my tablet emulator correctly.

## Screenshots of UI
| Before | After |
:-------------------------:|:-------------------------:
header_before | header_after

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by Mehul-Bisht** [comment]
@ARJUPTA @rt4914 kindly review the changes

**Review by Arjupta — CHANGES_REQUESTED** [review]
@Mehul-Bisht -
- Refer the issue number in the description
- Screenshot of UI before and after the changes . Also for the tests that are passing

Can you explain the which part of the issue did you tried to solve. because there is only one problem mentioned in the #2322 ie of small clickable region

**Comment by Mehul-Bisht** [comment]
> Can you explain the which part of the issue did you tried to solve. because there is only one problem mentioned in the #2322 ie of small clickable region

@ARJUPTA the previous responses header item had an accessibility issue with its touch target height that I fixed in this PR for mobile layouts, but not for the tablet layout. I have also updated the PR description.

**Comment by Arjupta** [comment]
> > Can you explain the which part of the issue did you tried to solve. because there is only one problem mentioned in the #2322 ie of small clickable region
> 
> @ARJUPTA the previous responses header item had an accessibility issue with its touch target height that I fixed in this PR for mobile layouts, but not for the tablet layout. I have also updated the PR description.

Okay taking a look. Meanwhile can you resolve the conflict

**Comment by Mehul-Bisht** [comment]
> Okay taking a look. Meanwhile can you resolve the conflict

The conflict seems not to be a result of changes made in this PR. Is there anything you can suggest that I may try?

**Review by Arjupta — COMMENTED** [review]
@rt4914 The results are like this for the mobile layout
|Before|After|
|:-:|:-:|
|Screenshot_1613240390|Screenshot_1613240871|
|Screenshot_1613240453|Screenshot_1613240845|

but do we need to change the padding dimension to what @Mehul-Bisht suggested or we should keep earlier ones as they were more descriptive

**Comment by Arjupta** [comment]
> > Okay taking a look. Meanwhile can you resolve the conflict
> 
> The conflict seems not to be a result of changes made in this PR. Is there anything you can suggest that I may try?

The conflict is about having duplicate dimensions, try to merge your code locally to develop and then see the files which show conflict. You can learn about resolving conflicts from any video.

**Comment by Mehul-Bisht** [comment]
@ARJUPTA I've resolved the conflicts, kindly discuss with Rajat and let me know about the final decision

File: app/src/main/res/values/dimens.xml [hunk 1]
```diff
@@ -176,9 +176,9 @@
   <dimen name="previous_button_exploration_view_padding_start">24dp</dimen>
   <dimen name="previous_button_question_view_padding_start">24dp</dimen>
   <dimen name="previous_responses_item_exploration_split_view_margin_top">24dp</dimen>
-  <dimen name="previous_responses_item_exploration_view_margin_top">24dp</dimen>
+  <dimen name="previous_responses_item_exploration_view_margin_top">0dp</dimen>
   <dimen name="previous_responses_item_question_split_view_margin_top">24dp</dimen>
-  <dimen name="previous_responses_item_question_view_margin_top">24dp</dimen>
+  <dimen name="previous_responses_item_question_view_margin_top">0dp</dimen>
```
Comment (rt4914) [inline app/src/main/res/values/dimens.xml:181]: If there is a padding / margin of `0dp` remove that and use `space_0dp` where it was used in code.

**Review by rt4914 — CHANGES_REQUESTED** [review]
@Mehul-Bisht PTAL.

File: app/src/main/res/values/dimens.xml (hunk 1 above)
Comment (rt4914) [reply app/src/main/res/values/dimens.xml:181]: Also merge with latest develop

Comment (Mehul-Bisht) [reply app/src/main/res/values/dimens.xml:181]: @rt4914 Ok, and should I let this dimension be 0dp now or revert it back to 24dp?

Comment (Arjupta) [reply app/src/main/res/values/dimens.xml:181]: @rt4914 so are we following that `space_0dp` suggestion which we concluded while renaming all the dimensions during SLOP

Comment (Arjupta) [reply app/src/main/res/values/dimens.xml:181]: > @rt4914 Ok, and should I let this dimension be 0dp now or revert it back to 24dp?

You just have to use space_0dp in place of a named dimension with a value of 0dp. So don't revert back.

Comment (rt4914) [reply app/src/main/res/values/dimens.xml:181]: Basically all dimens values should be of some value other than 0dp and whereever you want to use `0dp` use `space_0dp`.

**Review by Arjupta — CHANGES_REQUESTED** [review]
@Mehul-Bisht as `previous_responses_item_exploration_view_margin_top` and `previous_responses_item_question_view_margin_top` are replaced with space_0dp, hence they can be removed from dimens.xml

**Comment by Mehul-Bisht** [comment]
>@Mehul-Bisht as previous_responses_item_exploration_view_margin_top and previous_responses_item_question_view_margin_top are replaced with space_0dp, hence they can be removed from dimens.xml

@ARJUPTA though i have not made changes in this PR for tablet layouts, should I still consider removing them from the mobile layouts?

**Comment by Arjupta** [comment]
> > @Mehul-Bisht as previous_responses_item_exploration_view_margin_top and previous_responses_item_question_view_margin_top are replaced with space_0dp, hence they can be removed from dimens.xml
> 
> @ARJUPTA though i have not made changes in this PR for tablet layouts, should I still consider removing them from the mobile layouts?

Then it will better if you include the similar changes for a Tablet Layout too and ask us for a review if you are not able to test it

**Comment by Mehul-Bisht** [comment]
## Update:

## Portrait Mode for Tablet:
| Before |
:-------------------------:
screenshot-2021-03-01_03 15 51 599
| After |
screenshot-2021-03-01_03 29 01 176

## Landscape Mode for Tablet:

There is no issue with landscape mode for tablet in the emulator that I used, here is the result for me
screenshot-2021-03-01_03 16 29 19

**Comment by Mehul-Bisht** [comment]
@ARJUPTA @rt4914 Please check the changes for tablet layouts, I was able to setup the emulator for tablet.

File: app/src/main/res/values-sw600dp-port/dimens.xml [hunk 2]
```diff
@@ -54,6 +54,7 @@
   <dimen name="previous_responses_item_exploration_view_margin_top">24dp</dimen>
   <dimen name="previous_responses_item_question_split_view_margin_top">24dp</dimen>
   <dimen name="previous_responses_item_question_view_margin_top">24dp</dimen>
+  <dimen name="previous_response_header_padding_top">24dp</dimen>
```
Comment (Arjupta) [inline app/src/main/res/values-sw600dp-port/dimens.xml]: As these values are not different from the ones in `values/dimens.xml` so they are not needed to be repeated here. Also we should remove the top_margins as they are already set to be space_0dp

**Review by Arjupta — CHANGES_REQUESTED** [review]
@Mehul-Bisht

**Comment by Mehul-Bisht** [comment]
@ARJUPTA PTAL now

File: app/src/main/res/values/dimens.xml (hunk 1 above)
Comment (rt4914) [reply app/src/main/res/values/dimens.xml:181]: This does not seem to be solved yet.

**Review by rt4914 — CHANGES_REQUESTED** [review]
@Mehul-Bisht Can you please add before and after screenshot with exact combination as a proof that this PR does not change any margins in overall. Mobile portrait, landscape tablet portrait and landscape.

**Comment by Mehul-Bisht** [comment]
## Mobile mode

##  Portrait mode
| Before | After |
:-------------------------:|:-------------------------:
mobile_port_before | mobile_port_after

## Landscape mode
| Status |
:-------------------------:
## Before 
mobile_landscape_before
## After 
mobile_landscape_after

## Tablet mode

##  Portrait mode
| Status | 
:-------------------------:
## Before
tablet_port_before 

## After
tablet_port_after

## Landscape mode
| Status |
:-------------------------:
## Before 
tablet_land_before

## After
tablet_land_after

**Comment by Mehul-Bisht** [comment]
> @Mehul-Bisht Can you please add before and after screenshot with exact combination as a proof that this PR does not change any margins in overall. Mobile portrait, landscape tablet portrait and landscape.

@rt4914 PTAL

**Comment by rt4914** [comment]
@Mehul-Bisht There is some issue with this PR because of which it is showing 100+ files changed and also checks are not running. Can you try merging with develop again?

**Comment by Mehul-Bisht** [comment]
@rt4914 I had merge conflicts but I was unable to view them in my local branch, so I pulled changes into develop and got changes from latest develop. Kindly PTAL, the checks have started running now.

**Comment by rt4914** [comment]
@Mehul-Bisht The PR looks good but you have missed one change. Please take a look at this: https://github.com/oppia/oppia-android/pull/2703/files#r576944928

**Comment by Mehul-Bisht** [comment]
@rt4914 I had a doubt, if there is a dimension that is being used by one of the tablet orientations only and not by the other 3 (mobile landscape and portrait and the other tablet orientation) then how should I be declaring it for that single orientation? Since the IDE warns me with `dimen has no declaration in the base values folder; this can lead to crashes when the resource is queried in a configuration that does not match this qualifier`. For other orientations it will be using `space_0dp` and hence does not require to be declared in the base values folder.

**Comment by rt4914** [comment]
> @rt4914 I had a doubt, if there is a dimension that is being used by one of the tablet orientations only and not by the other 3 (mobile landscape and portrait and the other tablet orientation) then how should I be declaring it for that single orientation? Since the IDE warns me with `dimen has no declaration in the base values folder; this can lead to crashes when the resource is queried in a configuration that does not match this qualifier`. For other orientations it will be using `space_0dp` and hence does not require to be declared in the base values folder.

@Mehul-Bisht I recently encountered this issue and therefore we will need to create the same name in `values/dimens.xml` file too with `0dp` value.

**Comment by anandwana001** [comment]
@Mehul-Bisht any update on this?

**Comment by Mehul-Bisht** [comment]
> @Mehul-Bisht any update on this?

Yes, I will be making the suggested changes as soon as possible

**Comment by Mehul-Bisht** [comment]
@rt4914 I seem to have lost track of what that issue here was, regarding naming dimens in all files to `0dp` which have a `0dp` value instead of some value. I think the portrait mode for tablet doesn't have `previous_responses_item_exploration_view_margin_top` set to `0dp` yet. Was that the issue? 

Also, since I'm building from VirtualBox Ubuntu I won't be able to test for tablets, only on my real device (mobile). Kindly someone test changes for tablet layouts.

**Comment by rt4914** [comment]
@Mehul-Bisht You can leave this issue. As I was working on this screen it became evident that UI depends upon other items too so its not that straight forward. Closing this now.

**Pull request closed without merging**